atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/AlterTableMetadataReferenceTest.java

```diff
@@ -0,0 +1,97 @@
+ */
+
+package com.palantir.atlasdb.keyvalue.dbkvs;
+
+import static com.palantir.logsafe.testing.Assertions.assertThatLoggableException;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.palantir.atlasdb.config.AtlasDbConfigs;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+public class AlterTableMetadataReferenceTest {
+    private static final ObjectMapper OBJECT_MAPPER = AtlasDbConfigs.OBJECT_MAPPER;
+    public static final TableReference PRESENT_TABLE_REFERENCE = TableReference.create(Namespace.create("test"), "foo");
+    public static final String PRESENT_PHYSICAL_TABLE_NAME = "testfoo";
+    public static final String MISSING_PHYSICAL_TABLE_NAME = "asdasdasd";
+    public static final TableReference MISSING_TABLE_REFERENCE =
+            TableReference.create(Namespace.EMPTY_NAMESPACE, "foo");
+
+    @Test
+    public void canDeserialiseMixOfTableReferenceAndPhysicalTableName() throws JsonProcessingException {
+        String config = "- namespace:\n   name: test\n  tablename: foo\n- physicalTableName: testfoo";
+        List<AlterTableMetadataReference> references = OBJECT_MAPPER.readValue(config, new TypeReference<>() {});
+        assertThat(references)
+                .contains(
+                        ImmutableTableReferenceWrapper.of(PRESENT_TABLE_REFERENCE),
+                        ImmutablePhysicalTableNameWrapper.of(PRESENT_PHYSICAL_TABLE_NAME));
+    }
+
+    @Test
+    public void throwsIfFailToDeserialise() throws JsonProcessingException {
+        String config = "namespace: test\ntablename: foo";
+        assertThatThrownBy(() -> OBJECT_MAPPER.readValue(config, AlterTableMetadataReference.class))
+                .rootCause()
+                .isInstanceOf(SafeRuntimeException.class)
+                .satisfies(exc -> assertThatLoggableException((SafeRuntimeException) exc)
+                        .hasLogMessage(
+                                "The alterTablesOrMetadataToMatchAndIKnowWhatIAmDoing value is specified incorrectly."
+                                        + " Please either specify a physical table name via `physicalTableName:"
+                                        + " physicalTableName`, or as a table reference via \n"
+                                        + "`namespace:\n"
+                                        + "  name: namespace\n"
+                                        + "tablename: tableName")
+                        .hasExactlyArgs(SafeArg.of("node", Map.of("namespace", "test", "tablename", "foo"))));
```

Comment:
sure. This seems a bit overkill (I'm guessing you're checking that the configured object is returned as an arg in JSON-deserialised format), but fine.

Reply:
Yeah it's a little, just trying to make it easier to debug when we can't see the config but they downdraft the full log line.